compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -65,6 +77,56 @@ func (n *NodeInventoryCollector) Scan(nodeName string) (*storage.NodeInventory,
 	return m, nil
 }
 
+// callAnalyzer isolates the call to Analyze to ensure we recover from any panic that might occur.
+// Analyze runs on a potentially unknown live file system, so it might be prone to unexpected behavior.
+// The unexpected behavior of analyzerFunction that we protect against include:
+// (1) panicking, (2) taking too long to execute (and potentially consuming node resources).
+func (n *NodeInventoryCollector) callAnalyzer(nodeName string) (*nodes.Components, error) {
+
+	type analyzerResult struct {
+		components *nodes.Components
+		err        error
+	}
+
+	results := make(chan analyzerResult)
+	ctx, cancel := context.WithTimeout(context.Background(), n.analyzerCallDeadline)
+	defer cancel()
+
+	go func() {
+		defer func(c chan analyzerResult) {
+			r := recover()
+			if r != nil {
+				log.Warnf("Analyzer call panicked: %v", r)
+				select {
+				case c <- analyzerResult{
+					components: nil,
+					err:        fmt.Errorf("call to Analyzer function panicked: %v", r),
+				}:
+				case <-ctx.Done():
+					return
+				}
+
+			}
+			close(c)
+		}(results)
```

Comment:
@Maddosaurus Maybe worth looking into https://github.com/stackrox/stackrox/blob/master/pkg/safe/run.go#L38

Reply:
Yes, we saw that. I am afraid we need some more customization to handle timeouts & panics together, but I will give it a try.

Reply:
I took some parts of that pkg and refactored a bit. Unfortunately, I couldn't reuse it 1:1 as the function we need here was unexported.